opensrp-chw-core/src/test/java/org/smartregister/chw/core/interactor/CoreAncMemberProfileInteractorTest.java

```diff
@@ -0,0 +1,134 @@
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class CoreAncMemberProfileInteractorTest extends BaseUnitTest {
+
+    private final Visit lastVisit = new Visit();
+    private final Set<Task> taskList = new HashSet<>();
+    private CoreAncMemberProfileInteractor interactor;
+    private MemberObject memberObject;
+    @Mock
+    private BaseAncMemberProfileContract.InteractorCallBack interactorCallBack;
+    @Mock
+    private AncLibrary ancLibrary;
+    @Mock
+    private VisitRepository visitRepository;
+    @Mock
+    private CoreChwApplication chwApplication;
+    @Mock
+    private TaskRepository taskRepository;
+    @Mock
+    private ChwTaskRepository chwTaskRepository;
+    @Mock
+    private AncMemberProfileContract.InteractorCallBack callback;
+
+    private AppExecutors appExecutors = new AppExecutors(Executors.newSingleThreadExecutor(), Executors.newSingleThreadExecutor(), Executors.newSingleThreadExecutor());
+
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        Context context = Context.getInstance();
+        CoreLibrary.init(context);
+
+        //Auto login by default
+        context.session().start(context.session().lengthInMilliseconds());
+
+        interactor = Mockito.mock(CoreAncMemberProfileInteractor.class, Mockito.CALLS_REAL_METHODS);
+
+        memberObject = Mockito.mock(MemberObject.class);
+        memberObject.setPregnancyRiskLevel("Low");
+        memberObject.setBaseEntityId("some-base-entity-id");
+        memberObject.setFamilyName("Some Family Name");
+
+        ReflectionHelpers.setStaticField(AncLibrary.class, "instance", ancLibrary);
+    }
+
+    @Test
+    public void getPregnancyRiskDetailsTest() {
+        interactor.getPregnancyRiskDetails(memberObject);
+        Assert.assertNull(memberObject.getPregnancyRiskLevel());
+    }
+
+    @Test
+    public void getLastVisitDateTest() {
+        Date lastVisitDate = new Date();
+
+        doReturn(visitRepository).when(ancLibrary).visitRepository();
+        when(visitRepository.getLatestVisit(memberObject.getBaseEntityId(), "ANC Home Visit")).thenReturn(lastVisit);
+        lastVisit.setDate(lastVisitDate);
+        Assert.assertEquals(interactor.getLastVisitDate(memberObject), lastVisitDate);
+    }
+
+    @Test
+    public void getClientTasksTest() {
+        doReturn(taskRepository).when(chwApplication).getTaskRepository();
+        String planId = "some plan Id";
+        String baseEntityId = "some baseEntityId";
+        when(chwTaskRepository.getReferralTasksForClientByStatus(planId, baseEntityId, "Referred")).thenReturn(taskList);
+        Task task = new Task();
+        taskList.add(task);
+        callback.setClientTasks(taskList);
+        Mockito.doNothing().when(callback).setClientTasks(taskList);
+        interactor.getClientTasks(planId, baseEntityId, callback);
+        Mockito.verify(interactor).getClientTasks(planId, baseEntityId, callback);
```

Comment:
Won't this always evaluate to true since calling `getClientTasks` in line 106 means that verifying it was called will always be true? 🤔

Reply:
Let me change this